src/main/java/org/plumelib/util/StringsPlume.java

```diff
@@ -1049,24 +1049,44 @@ public static String mapToStringAndClass(Map<?, ?> m) {
 
   /**
    * Return either "n <em>noun</em>" or "n <em>noun</em>s" depending on n. Adds "es" to words ending
-   * with "ch", "s", "sh", or "x".
+   * with "ch", "s", "sh", or "x", adds "ies" to words ending with "y" when the previous letter is
+   * consonant, adds "ves" to words ending in "f" or "fe".
    *
    * @param n count of nouns
    * @param noun word being counted
    * @return noun, if n==1; otherwise, pluralization of noun
    */
   @SideEffectFree
   public static String nplural(int n, String noun) {
+    if (noun.length() < 2) {
+      return "null";
```

Comment:
I think clients may want to use a noun to have a length of 1.
Suppose some code is counting the number of x characters in a string.
It might call `nplural(count, "x")`.

More generally, I don't think it is desirable to arbitrarily forbid an argument of length 1, since there is no good reason for it (except convenience in the implementation).

I agree that the current code that sets `penultimateLetter` must be adjusted, but it's better to make the implementation a bit more complex than to make the specification more complex and less useful.

Reply:
Ok it makes sense. I think that the simplest solution would be to put an extra condition in the if statement. Something like that: if the length of the noun is 1, then boolean letter = false. And the variable letter could be used in the if statement later: if(lastLetter == 'y') && letter && .... I do not know if this is the best solution but is the simplest i could think.

Reply:
That will work.  I wouldn't name a boolean `letter`, but rather `lengthIs1` or the like.
Another possibility is to set `penultimateLetter` to an arbitrary value if the string has length 1 (and possibly add tests for it later).

Reply:
I wouldn't name a boolean like that, i just chose a name for the example.  The only tricky part to your suggestion is that we need to define carefully the arbitrary value.